### DISCUSSION TITLE
[Supersim Interoperability] doubt in implementation of message passing using Supersim

### DISCUSSION THREAD

**Discussion by Kaushal-205** [ethereum-optimism/developers#663] [Application Development]
### Did you check the documentation?

- [x] Yes
- [ ] No

### What type of issue are you experiencing?

- [ ] Error
- [ ] Feature Request
- [X] General Question

### Issue Description

I'm creating a contract which Disperses the ERC20 token to multiple recipient on destination chain.
I've also created Script in which mints the superchainWETH tokens and send it via crossdomainmessenger, 
It's successfully executing the 
"transferTokens" function but "receiveTokens" is not called.

for better understanding, I'm sending 3 WETHs from Chain 901 to 902 with the message containing Two EOA addresses with amount 1 WETH and 2 WETH respectively, 
expected result should be on chain 902, WETH should be dispersed to the EOA accounts as per the received message.

and there should not be extra function call after calling Transfertokens function on chain 901 by user (there can be internal function calls)

here's the github repo of that  : [https://github.com/Kaushal-205/SmartDisperse-supersim]

please let me know, if you need any other thing in understanding or running the script

Thank you

### Steps to Reproduce

1. Start the anvil node of supersim using 

```supersim --interop.autorelay```

2. install required library using 

```forge install```

3. create .env file according to the given .env-example file

4. Run the forge script
```forge script script/SmartDisperse.s.sol``

### Are you using a specific library or SDK?

- [ ] OP-SDK
- [ ] Viem
- [X] Other
- [ ] Not Sure

### Are you using the latest version of the library/SDK?

- [X] Yes
- [ ] No

### What type of support do you need?

- [X] Technical support for issue resolution
- [X] Feature implementation guidance
- [X] General advice or consultation
- [ ] Documentation or learning resources

**Comment by opfocus** [comment]
Due to my limited knowledge, I may not be able to answer your question accurately.
1. You are trying to relay a message using the` L2_TO_L2_CROSS_DOMAIN_MESSENGER`, right? Has that been successful?
2. Even if the message can be relayed, considering that you might not have minting permissions for WETH in your contract, it may not successfully receive the tokens.   
https://github.com/ethereum-optimism/optimism/blob/e8de7b191b0df2fd454fce6955bc5702e34bf297/packages/contracts-bedrock/src/L2/SuperchainWETH.sol#L67C1-L70C29
```
 function crosschainMint(address _to, uint256 _amount) external {
        if (msg.sender != Predeploys.SUPERCHAIN_TOKEN_BRIDGE) revert Unauthorized();

        _mint(_to, _amount);

        // Mint from ETHLiquidity contract.
        if (!IL1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) {
            IETHLiquidity(Predeploys.ETH_LIQUIDITY).mint(_amount);
        }

        emit CrosschainMint(_to, _amount);
    }
```
